### PR TITLE
Generated DisruptionCron client

### PR DESCRIPTION
**Pull request by diyarab**
## What does this PR do?

- [x] Adds new functionality
- [ ] Alters existing functionality
- [ ] Fixes a bug
- [ ] Improves documentation or testing

## Please briefly describe your changes as well as the motivation behind them:
This PR introduces a new client for `DisruptionCron`. The client was generated using [client-gen](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-api-machinery/generating-clientset.md), a tool provided by Kubernetes for generating typed clients for custom resources. 

After setting up `client-gen`, navigate to the `chaos-controller` directory in your terminal and reproduce the code by running the following command:
```bash
client-gen \
--input-base="github.com/DataDog/chaos-controller/api" \
--input="v1beta1" \
--clientset-name="v1beta1" \
--output-dir="$(pwd)/clientset" \
--output-pkg="github.com/DataDog/chaos-controller/clientset" \
--fake-clientset=false
```

## Code Quality Checklist

- [ ] The documentation is up to date.
- [x] My code is sufficiently commented and passes continuous integration checks.
- [x] I have signed my commit (see [Contributing Docs](../CONTRIBUTING.md)).

## Testing

- [ ] I leveraged continuous integration testing
    - [ ] by depending on existing `unit` tests or `end-to-end` tests.
    - [ ] by adding new `unit` tests or `end-to-end` tests.
- [ ] I manually tested the following steps:
    - `x`
    - [x] locally.
    - [ ] as a canary deployment to a cluster.

### CODE REVIEW

**Review by taihuynh167 — DISMISSED** [review]

File: clientset/v1beta1/disruption_interface_mock.go
Comment (aymericDD) [inline clientset/v1beta1/disruption_interface_mock.go:1]: Maybe we should deprecated this file if it's used by other projects and/or create/generate new mocks for the new client interfaces. WDYT?

Comment (aymericDD) [reply clientset/v1beta1/disruption_interface_mock.go:1]: To do that you just have to launch the `make generate-mocks` command 👍

**Review by aymericDD — DISMISSED** [review]
LGTM 👍 
I just let a comment about the mock part, and also it could be nice to have inside the make the command to generate the new clients and/or a documentation to know how to use client and how to generate them.

**Review by ptnapoleon — COMMENTED** [review]
Looks great! But just as Aymeric mentioned, could you include a `make` target to re-generate the clientset? And optionally, could you add another CI target to check if the clientset needs regenerated? The target can run your new `make` target, followed by `git diff`, and fail if any changes are detected

File: api/v1beta1/install/install.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024 Datadog, Inc.
+// Package install installs the experimental API group, making it available as
+// an option to all of the API encoding/decoding machinery.
+package v1beta1
```
Comment (aymericDD) [inline api/v1beta1/install/install.go:7]:
```suggestion
package install
```

Maybe, I'm wrong, but usually the name of the package is the same as the folder.

**Comment by diyarab** [comment]
> Looks great! But just as Aymeric mentioned, could you include a `make` target to re-generate the clientset? And optionally, could you add another CI target to check if the clientset needs regenerated? The target can run your new `make` target, followed by `git diff`, and fail if any changes are detected

**Comment by aymericDD** [comment]
Okay, no problem for me until we have ticket to treat this topic. 👍  
Can you complete the documentation by adding a `README` or modify an existing one to explain how to generate the clients? _It will avoid for new mates to make the same mistake as us and try to create them by hand_ 😉

**Comment by aymericDD** [comment]
PR which add the documentation https://github.com/DataDog/chaos-controller/pull/855

**Review by aymericDD — APPROVED** [review]
Good job 👌

**Pull request merged**